middlewares/osio/osio.go

```diff
@@ -99,85 +125,123 @@ func NewPreConfiguredOSIOAuth() *OSIOAuth {
 	if len(srvAccSecret) <= 0 {
 		panic("Missing SERVICE_ACCOUNT_SECRET")
 	}
-	return NewOSIOAuth(tenantURL, authURL, srvAccID, srvAccSecret)
+	return NewOSIOAuth(tenantURL, authURL, srvAccID, srvAccSecret, authTokenKey)
 }
 
-func NewOSIOAuth(tenantURL, authURL, srvAccID, srvAccSecret string) *OSIOAuth {
+func NewOSIOAuth(tenantURL, authURL, srvAccID, srvAccSecret, authTokenKey string) *OSIOAuth {
 	return &OSIOAuth{
 		RequestTenantLocation: CreateTenantLocator(http.DefaultClient, tenantURL),
-		RequestTenantToken:    CreateTenantTokenLocator(http.DefaultClient, authURL),
+		RequestTenantToken:    CreateTenantTokenLocator(http.DefaultClient, authURL, authTokenKey),
```

Comment:
`authTokenKey` passed to tenant_service as param rather tenat_service uses to `os.Getenv()`.  This is re-factoring and not related to oso_proxy_v2.